[NFC] Hide SourceFile::Decls

CodaFi opened the pull request:
In preparation for installing some stable paths infrastructure here,
hide access to the array of top-level decls.

CodaFi committed:
[NFC] Hide SourceFile::Decls

In preparation for installing some stable paths infrastructure here,
hide access to the array of top-level decls.

CodaFi commented:
@swift-ci please smoke test and merge

davidungar commented:
Nice!

davidungar reviewed (approved):
LGTM!

slavapestov commented on include/swift/AST/SourceFile.h:
```diff
@@ -125,11 +125,28 @@ class SourceFile final : public FileUnit {
-  /// The list of top-level declarations in the source file.
-  std::vector<Decl*> Decls;
+  /// Appends the given declaration to the end of the top-level decls list.
+  void addTopLevelDecl(Decl *d) {
```
Eventually this should set them all in one shot so that anything depending on getTopLevelDecls() is immutable.

CodaFi replied:
LLDB is going to be the last bastion of mutations here...

slavapestov reviewed (approved):
Nice cleanup!

CodaFi commented:
Urf, need LLDB patches.

CodaFi commented:
apple/llvm-project#525

@swift-ci smoke test

CodaFi commented:
apple/llvm-project#525

@swift-ci smoke test

CodaFi commented:
apple/llvm-project#525

@swift-ci smoke test

CodaFi commented:
apple/llvm-project#525

@swift-ci smoke test

CodaFi committed:
Introduce a special LLDB-only way to prepend decls

SwiftASTManipulator is installing declarations in the wrong order, and
currently fails a REPL test if the right API is used.

rdar://58355191 is tracking the removal of this entrypoint.

CodaFi commented:
Installed a workaround for LLDB and filed rdar://58355191 about removing it on their end.

CodaFi commented:
apple/llvm-project#525

@swift-ci smoke test

CodaFi commented:
apple/llvm-project#525

@swift-ci smoke test

CodaFi commented:
⛵️
